akka-cluster-tools/src/main/scala/akka/cluster/client/ClusterClient.scala

```diff
@@ -259,6 +336,8 @@ final class ClusterClient(settings: ClusterClientSettings) extends Actor with Ac
   var contacts = initialContactsSel
   sendGetContacts()
 
+  var contactsPublished = Set.empty[ActorSelection]
```

Comment:
we should not use `ActorSelection` here, better to keep the original `ActorPath` in a new `contactPaths: Set[ActorPath]` field and this would be `contactsPublished = Set.empty[ActorPath]`